dojo/tools/generic/parser.py

```diff
@@ -52,14 +53,30 @@ def _get_test_json(self, data):
         )
         test_internal.findings = list()
         for item in data.get("findings", []):
+            # remove endpoints of the dictionnary
+            unsaved_endpoints = None
+            if "endpoints" in item:
+                unsaved_endpoints = item["endpoints"]
+                del item["endpoints"]
+            # remove files of the dictionnary
+            unsaved_files = None
+            if "files" in item:
+                unsaved_files = item["files"]
+                del item["files"]
+            # remove vulnerability_ids of the dictionnary
+            unsaved_vulnerability_ids = None
+            if "vulnerability_ids" in item:
+                unsaved_vulnerability_ids = item["vulnerability_ids"]
+                del item["vulnerability_ids"]
+
             # check for required keys
             required = ['title', 'date', 'severity', 'description']
             missing = []
             for field in required:
-                if field not in data:
+                if field not in item.keys():
                     missing.append(field)
             if missing:
-                raise Exception(f"Required fields are missing: {missing}")
+                raise ValidationError(f"Required fields are missing: {missing}")
```

Comment:
Please use `ValueError` instead. Using a Django error could be misleading.
```suggestion
                raise ValueError(f"Required fields are missing: {missing}")
```

Reply:
I see your point. But if we will use only `ValueError`, the user will receive `HTTP-500` with 
```json
{
  "message": "Internal server error, check logs for details"
}
```
and he/she/they will have to contact the admin for logs to identify small mistakes.
In the case of `ValidationError`, the user will be able to see the error message in the response and identify the problem and solve it.

What about using `ValueError`s in parsers and "rewrap" them to `ValidationError` in (re)importer?